### PR TITLE
Adds a new mobility option for wizards, the Telegram Scepter, not inspired by any relic at all

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

This PR adds a new mobility option for wizards, the Telegram Scepter, a magical rod that teleports you anywhere you point it. 

But be warned, its method of teleportation is very volatile, putting you in a state of flux. While the scepter itself has no cooldown, using it while in flux will damage you severely and purge some of your chems. In fact, any form of magical teleportation will damage you while in flux. 

https://github.com/tgstation/tgstation/assets/51863163/b1008f6c-d724-4b7d-8e47-b70e2e770183

Non-wizards who attempt to utilize the wand will find themselves in a state of permanent flux, meaning any and all teleports are met with damage. It's also less precise. 

Also adds an admin-only version which has infinite range and applies no flux.

Any semblances to any other game is purely coincidental, I assure you.

Misc. Details: 

- I added the ability to pass sound datums to playsound, so it doesn't override custom sound settings you may want. 
- I moved the teleport signals out of the pure-sounding proc `check teleport`. 

## Why It's Good For The Game

I've always thought wizard mobility has been rather lacking, in that you generally only pick Jaunt.

image

So when someone threw an offhanded suggestion to make a certain item from a certain game in discord, my wheel started to turn.

And here we are.

- Teleporting anywhere you can see is very powerful (especially in conjunction with the Scrying Orb) so this makes it a very high potency item a wizard can pick up, almost like a targeted blink. 
- It is, however, offset by a few things: Namely, the flux mechanic putting a soft cooldown on teleporting (lest you kill yourself), and it being an item, meaning you may have to pick up Instant Summons for added safety. 

Overall, I thought it to be a very flavorful mobility option for wizards. Maybe it will shake up spell selection. Who knows?

## Changelog

:cl: Melbert
add: Wizards have a new mobility option available, the Telegram Scepter. The ability to travel anywhere you can see at the point of a wand... but at a price?
/:cl:

### CODE REVIEW

**Comment by mc-oofert** [comment]
https://terraria.fandom.com/wiki/Rod_of_Discord

**Review by Wallemations — APPROVED** [review]
Hey has anyone seen my lightsaber flyswatter anywhere

**Comment by Xkallubar** [comment]
Literally just the Rod of Discord from Terraria

**Comment by Jezzared** [comment]
> Any semblances to any other game is purely coincidental, I assure you.

Yall just do not get jokes do you

**Comment by Exester509** [comment]
> https://terraria.fandom.com/wiki/Rod_of_Discord

https://terraria.wiki.gg/wiki/Rod_of_Discord

The fandom wiki is no longer officially supported and might be outdated. Please do not spread misinformation over the internet

**Comment by mc-oofert** [comment]
> > https://terraria.fandom.com/wiki/Rod_of_Discord
> 
> https://terraria.wiki.gg/wiki/Rod_of_Discord
> 
> The fandom wiki is no longer officially supported and might be outdated. Please do not spread misinformation over the internet

Nuh uh

**Comment by Shadowflame909** [comment]
This looks cool!

File: code/modules/reagents/chemistry/holder/holder.dm [hunk 1]
```diff
@@ -333,6 +333,27 @@
...
+/datum/reagents/proc/remove_all_direct(percentage = 0.5)
+	if(!total_volume)
+		return 0
+	if(percentage <= 0)
+		stack_trace("non positive percentage passed to remove all reagents direct [percentage]")
+		return 0
+	var/list/cached_reagents = reagent_list
+	var/total_removed_amount = 0
+
+	for(var/datum/reagent/reagent as anything in cached_reagents)
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/holder/holder.dm:351]:
```suggestion
	var/total_removed_amount = 0

	for(var/datum/reagent/reagent as anything in reagent_list)
```
Frankly, it's already cached the moment you use it in such a loop iirc, and the file is already chock-full of this "cached_reagents" idiocy.

**Comment by carlarctg** [comment]
but why it remove chems

**Comment by Ghommie** [comment]
> but why it remove chems

He said it'd make an anti-counter for stuff like mute toxin, though, frankly, it's not all that much used.

**Comment by MrMelbert** [comment]
Because Pros: Purges stuff like Mute toxin or other poisons 
and Cons: Purges brute healing chems to stop free teleporting

**Comment by adamkad1** [comment]
Teamspeak wand when?

**Pull request merged**